lib/zipkin-tracer/rack/zipkin-tracer.rb

```diff
@@ -46,23 +47,17 @@ def annotate_plugin(span, env, status, response_headers, response_body)
     end
 
     def trace!(span, zipkin_env, &block)
-      trace_request_information(span, zipkin_env)
-      span.kind = Trace::Span::Kind::SERVER
       status, headers, body = yield
     ensure
+      trace_server_information(span, zipkin_env, status)
+
       annotate_plugin(span, zipkin_env.env, status, headers, body)
     end
 
-    def trace_request_information(span, zipkin_env)
-      tags = if !@config.record_on_server_receive.empty?
-        # if the user specified tags to record on server receive, use these no matter what
-        @config.record_on_server_receive
-      elsif !zipkin_env.called_with_zipkin_headers?
-        # if the request comes from a non zipkin-enabled source record the default tags
-        DEFAULT_SERVER_RECV_TAGS
-      end
-      return if tags.nil?
-      tags.each { |annotation_key, env_key| span.record_tag(annotation_key, zipkin_env.env[env_key]) }
+    def trace_server_information(span, zipkin_env, status)
+      span.kind = Trace::Span::Kind::SERVER
+      span.record_tag(Trace::Span::Tag::STATUS, status)
```

Comment:
you can consider doubling this as error so the span appears red

ex from java
```java

    // 1xx, 2xx, and 3xx codes are not all valid, but the math is good enough vs drift and opinion
    // about individual codes in the range.
    if (httpStatusInt < 100 || httpStatusInt > 399) {
      customizer.tag("error", String.valueOf(httpStatusInt));
    }
```

Reply:
Here is how we are doing it in client spans:
https://github.com/openzipkin/zipkin-ruby/blob/cc4e358dbe8c6dab58d1aa62331d89f0ea42557e/lib/zipkin-tracer/faraday/zipkin-tracer.rb#L75

Maybe we can have a method like this in the [Span class](https://github.com/openzipkin/zipkin-ruby/blob/master/lib/zipkin-tracer/trace.rb#L166):
```ruby
    STATUS_ERROR_REGEXP = /\A(4.*|5.*)\z/.freeze

    def record_status(status)
      record_tag(Tag::STATUS, status)
      record_tag(Tag::ERROR, status) if STATUS_ERROR_REGEXP.match(status)
    end

Reply:
Done